### ISSUE TITLE
process context error

### ISSUE THREAD

**Issue by suytingwan** [facebookresearch/wsd-biencoders#3]
Hi, I try to replicate the demo and have found some errors.
1. in the function _preprocess_context_  in biencoder.py, line 145 and line 175, tokenizations of 'CLS' and 'SEP' return 3 numbers each. Is it right to use them as a list? If so, there will be problems in later as Bert inputs.
2. I download the best model ckpt but the evaluation result on semeval 2007 dataset is 51.9. I use Pytorch 1.2.0 and transformers 2.6.0. Do I miss any important procedure?

**Comment by blvns** [comment]
Hi! I think the issues in (1) and (2) are related. This code was written is using an old version of the transformers package, which is the pytorch_transformers package with version 1.1.0. It looks like tokenization was changed in a newer version of the code, which could be why you are getting the mis-tokenizations of the special tokens and an unexpected score with the checkpointed model.

**Comment by blvns** [comment]
I'm closing this issue, but let me know if you have any further issues!

**Comment by suytingwan** [comment]
> I'm closing this issue, but let me know if you have any further issues!

Thank you. I have solved the tokenization problem by changing the pytorch_transformers to version 1.1.0. However, the performance of ALL dataset is still 77.6. I have only TITANX at hand so I set the context batch size=1. I have changed the code to let gradient backward every 4 context. Do this influence the replicating? I am not sure. Still trying.

**Comment by blvns** [comment]
Sorry I didn't see this earlier. It's unlikely that the gradient accumulation itself would make a difference in replicating the model, but I've found that changes to the code and using different hardware can affect the random seed, which will make the model train differently. I wouldn't expect to replicate the exact same numbers with a different setting like that.
